src/matvis/cpu/matprod.py

```diff
@@ -50,3 +50,41 @@ def compute(self, z: np.ndarray, out: np.ndarray) -> np.ndarray:
             out[i] = z[ai].conj().dot(z[aj].T)
 
         return out
+
+
+class CPUMatChunk(MatProd):
```

Comment:
Another thing we might want to have in the `setup()` for this is to check that all the pairs in `antpairs` also exist in the `matsets`

Reply:
I have added a function that checks for this and throws an error if all the antpairs are not included somewhere in matsets.  The check only happens if the user selects the matrix chunking method.